src/runtime/HalideRuntime.h

```diff
@@ -379,8 +379,9 @@ typedef enum halide_type_code_t
 {
     halide_type_int = 0,   //!< signed integers
     halide_type_uint = 1,  //!< unsigned integers
-    halide_type_float = 2, //!< floating point numbers
-    halide_type_handle = 3 //!< opaque pointer type (void *)
+    halide_type_float = 2, //!< IEEE floating point numbers
+    halide_type_handle = 3, //!< opaque pointer type (void *)
+    halide_type_bfloat = 4, //!< floating point numbers in the bfloat format
```

Comment:
I know this is almost certainly necessary, but I predict we'll find a surprisingly long tail of code that will need to be tweaked for the addition of a new Typecode.

Reply:
Probably worth commenting on that to be sure no one in the future tries to "optimize" it

Reply:
Not sure what you mean - how would one "optimize" it?

Reply:
whoops, that response went on the wrong line. I meant, a comment on the line with the four-in-a-row-cast might be useful in case someone in the future thinks it's redundant.